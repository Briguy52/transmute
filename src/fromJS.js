// @flow
import { fromJS as fromJSBase } from 'immutable';

/**
 * A version of Immutable.fromJS that drops all but the first argument for
 * compatibility with other transmute functions like `map`.
 *
 * @example
 * fromJS({items: [1, 2, 3]})
 * // returns Map { items: List [ 1, 2, 3 ] }
 *
 * @param {any} json
 * @return {?Iterable}
 */
export default function fromJS<T>(json: T) {
  return fromJSBase(json);
}
