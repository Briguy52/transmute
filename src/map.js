import curry from "./curry";
import { Iterable } from "immutable";
import { map } from "./protocols/Reducable";

map.implement(Array, (mapper, arr) => arr.map(mapper));

map.implementInherited(Iterable, (mapper, iter) => iter.map(mapper));

map.implement(Object, (mapper, obj) => {
  const result = {};
  const keys = Object.keys(obj);
  const len = keys.length;
  for (let i = 0; i < len; i++) {
    const key = keys[i];
    result[key] = mapper(obj[key], key, obj);
  }
  return result;
});

/**
 * Create a new Iterable by applying `mapper` to each item in `subject`.
 *
 * @example
 * // returns List [ 2, 3, 4 ]
 * map(
 *   (val) => val + 1,
 *   List.of(1, 2, 3)
 * );
 *
 * @param  {Function} mapper applied to each item in `subject`.
 * @param  {Iterable} subject the Iterable to map.
 * @return {Iterable} with each value of `subject` updated with mapper.
 */
export default curry(map);
