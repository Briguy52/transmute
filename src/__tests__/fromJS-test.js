import fromJS from "../fromJS";
import { List } from "immutable";
import map from "../map";

describe("transmute/fromJS", () => {
  it("converts json to Immutable", () => {
    expect(fromJS({ nums: [1, 2, 3] })).toMatchSnapshot();
  });

  it("works with `map`", () => {
    const original = List.of({ one: 1 }, { two: 2 }, { three: 3 });
    expect(map(fromJS, original)).toMatchSnapshot();
  });
});
