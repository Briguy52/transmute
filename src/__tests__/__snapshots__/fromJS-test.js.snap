// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/fromJS converts json to Immutable 1`] = `
Immutable.Map {
  "nums": Immutable.List [
    1,
    2,
    3,
  ],
}
`;

exports[`transmute/fromJS works with \`map\` 1`] = `
Immutable.List [
  Immutable.Map {
    "one": 1,
  },
  Immutable.Map {
    "two": 2,
  },
  Immutable.Map {
    "three": 3,
  },
]
`;
